File: A2/HobbyAnimals/Animals/Hamster.cs
namespace HobbyAnimals;

//Primary Constructor inherited from Animal class
public class Hamster(string name, int exhilaration, char type) : Animal(name, exhilaration, type)
{
    public override void changeEx(Mood mood)
    {
        switch (mood)
        {
            case Mood.joyful:
                increaseEx(2);
                break;
            case Mood.usual:
                decreaseEx(3);
                break;
            case Mood.blue:
                decreaseEx(5);
                break;
        }
    }
}

File: A2_py/Animals/Hamster.py
class Hamster(Animal):
    def __init__(self, name: str, ex: int, type:str):
        self.name = name
        self.ex = ex
        self.type = type
    def changeEx(self, mood : Mood):
        match mood:
            case Mood.JOYFUL:
                self._increaseEx(2)
            case Mood.USUAL:
                self._decreaseEx(3)
            case Mood.BLUE:
                self._decreaseEX(5)
